frontend/test/metabase/scenarios/collections/permissions.cy.spec.js

```diff
@@ -440,17 +440,40 @@ describe("collection permissions", () => {
         cy.signInAsAdmin();
       });
 
-      it.skip("shouldn't record history steps when there was no diff (metabase#1926)", () => {
+      it("shouldn't render revision history steps when there was no diff (metabase#1926)", () => {
         cy.signInAsAdmin();
         cy.createDashboard("foo").then(({ body }) => {
           visitAndEditDashboard(body.id);
         });
+
         // Save the dashboard without any changes made to it (TODO: we should probably disable "Save" button in the first place)
         saveDashboard();
-        // Take a look at the generated history - there shouldn't be anything other than "First revision" (dashboard created)
-        cy.icon("ellipsis").click();
-        cy.findByText("Revision history").click();
-        cy.findAllByRole("button", { name: "Revert" }).should("not.exist");
+        cy.icon("pencil").click();
+        saveDashboard();
+
+        openRevisionHistory();
+
+        const revisionHistoryEntries = cy
+          .findAllByText("Bobby Tables")
+          .parent();
+
+        // This accounts for:
+        // 1. First revision.
+        // 2. added a card.
+        // 3. rearranged the cards. (TODO: consider fixing this as cards we never rearranged)
```

Comment:
The test is failing because we never actually get to this chain of events.
The original intent seems right and it seems to prove you solved this issue, but your final assertion is wrong. Let's take a look at the steps that do happen:
1. We create a new dashboard
2. Visit it and enter editing more
3. Save it (shouldn't generate a diff - it doesn't. Bingo!)
4. Enter editing mode again
5. Save it again without any real changes (again, no diff as expected)

So, the final assertion should be that no revision buttons are present at all and there is only only one history entry (First revision).

We can add an additional (version of this) test for the scenario when there's a card added if you want to cover that as well.

Reply:
Oh, that's one thing that I noticed and CAN'T figure out why, but...
Only and only when you first open Cypress GUI it runs the same thing over and over again, and doesn't even notice the changes. If you stop the test from here (see screenshot)
![image](https://user-images.githubusercontent.com/31325167/117463226-b9d5b600-af4f-11eb-8c97-79668e8e3952.png)
and then rerun the same file, it will start updating on every single change. It's annoying and undocumented so that's probably what happened to you.